components/dashboard/src/components/SelectWorkspaceClassComponent.tsx

```diff
@@ -17,36 +17,34 @@ interface SelectWorkspaceClassProps {
 }
 
 export default function SelectWorkspaceClassComponent(props: SelectWorkspaceClassProps) {
-    const [workspaceClasses, setWorkspaceClasses] = useState<SupportedWorkspaceClass[]>();
-    useEffect(() => {
-        getGitpodService().server.getSupportedWorkspaceClasses().then(setWorkspaceClasses);
-    }, []);
-    const getElements = useMemo(() => {
-        return () => {
-            if (!workspaceClasses) {
-                return [];
-            }
-            return [
-                ...workspaceClasses.map(
-                    (c) =>
-                        ({
-                            id: c.id,
-                            element: <WorkspaceClassDropDownElement wsClass={c} />,
-                            isSelectable: true,
-                        } as DropDown2Element),
-                ),
-            ];
-        };
-    }, [workspaceClasses]);
+    const workspaceClasses = useWorkspaceClasses();
+    const elements = useMemo(() => {
```

Comment:
Noob Q, why do we need the `memo` when we're already using the cache on `useWorkspaceClasses`?

Reply:
React runs the main function frequently (on every rerender) but would run what is in `useMemo` only if the dependency array has changed. So we basically cache the creation of the DropDownElement objects which included the WorkspaceClassDFropDownElement and its children. It's IMHO not really necessary and in many cases react users are prematurely optimizing here. This case is actually one of those places where it makes more sense.

Reply:
this caching using the dependency array works the same for `useEffect` and `useCallback` btw. They only get executed when the dependency array has changed.